boa/src/builtins/object/gcobject.rs

```diff
@@ -99,11 +106,12 @@ impl GcObject {
     // <https://tc39.es/ecma262/#sec-ecmascript-function-objects-call-thisargument-argumentslist>
     pub fn call(&self, this: &Value, args: &[Value], ctx: &mut Interpreter) -> Result<Value> {
         let this_function_object = self.clone();
-        let object = self.borrow();
-        if let Some(function) = object.as_function() {
+        let f_body: FunctionBody = if let Some(function) = self.borrow().as_function() {
```

Comment:
```suggestion
        let body = if let Some(function) = self.borrow().as_function() {
```

If type annotation is not needed its better to leave it to the compiler.